pkg/controller/network_policy.go

```diff
@@ -802,62 +818,116 @@ func (cont *AciController) getServiceAugmentBySubnet(subj apicapi.ApicObject,
 				servicekey+": ", err.Error())
 			continue
 		}
-		endpointsobj, _, err := cont.endpointsIndexer.GetByKey(servicekey)
-		if err != nil {
-			logger.Error("Could not lookup endpoints for "+
-				servicekey+": ", err.Error())
-			continue
-		}
-		if serviceobj == nil || endpointsobj == nil {
+		if serviceobj == nil {
 			continue
 		}
 		service := serviceobj.(*v1.Service)
-		endpoints := endpointsobj.(*v1.Endpoints)
-
-		for _, svcPort := range service.Spec.Ports {
-			if prs.port != nil &&
-				(svcPort.Protocol != *prs.port.Protocol ||
-					svcPort.TargetPort.String() !=
-						prs.port.Port.String()) {
-				// egress rule does not match service target port
+		if !cont.endPointSliceEnabled {
```

Comment:
Can we avoid this check everywhere in the code? We should define a go interface or function pointer that is setup at init and does the appropriate thing.

Reply:
Sure will create interface class for all this functions